src/modules/roc_sndio/backend_dispatcher.cpp

```diff
@@ -70,7 +70,10 @@ BackendDispatcher::BackendDispatcher()
 #endif // ROC_TARGET_SOX
 }
 
-void BackendDispatcher::set_frame_size(size_t frame_size) {
+void BackendDispatcher::set_frame_size(core::nanoseconds_t frame_length,
+                                       size_t sample_rate,
+                                       packet::channel_mask_t channels) {
+    size_t frame_size = packet::ns_to_size(frame_length, sample_rate, channels);
```

Comment:
Let's move this calculation even deeper, inside SoxBackend::set_frame_size(). BackendDispatcher is a high-level component and it'd be nice to hide these details from it.

Reply:
done.